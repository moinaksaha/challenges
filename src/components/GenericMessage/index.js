// @flow

// Necessary Imports
import React from 'react';
import PropTypes from 'prop-types';

// Styled components 
// Reference - https://github.com/styled-components/styled-components
import styled, { ThemeProvider } from 'styled-components';

// Styled Components
const MessageWrapper = styled.div`
    color: ${props => props.theme.main};
`;

MessageWrapper.defaultProps = {
  theme: {
    main: 'blue',
  },
};

// Dynamically deciding the theme based on props
const getMessageTheme = (type) => {
  switch (type) {
    case 'error':
      return {
        main: '#ff4646',
      }
    case 'success':
      return {
        main: '#429845',
      }
    case 'warning':
      return {
        main: '#FF9800',
      }
    case 'inProgress':
      return {
        main: '#648cf3',
      }
    default:
      return {
        main: '#429845',
      }
  }
};

const GenericMessage = ({ type, message }) => (
  <ThemeProvider theme={getMessageTheme(type)}>
    <MessageWrapper>
      {message}
    </MessageWrapper>
  </ThemeProvider>
);

// Proptype Validations
GenericMessage.propTypes = {
  type: PropTypes.string,
  message: PropTypes.string.isRequired,
};

// Default Props
GenericMessage.defaultProps = {
  type: 'success',
  message: 'Some Message',
};

export default GenericMessage


