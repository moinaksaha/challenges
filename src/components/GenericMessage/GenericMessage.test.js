import React from 'react'
import GenericMessage from './index.js'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Component: <GenericMessage />', () => {

  test('Render Status: Success', () => {
    const wrapper = shallow(<GenericMessage />);
    expect(wrapper.exists()).toBe(true);
  })

  test('Snapshot: Enzyme', () => {
    const tree = shallow(<GenericMessage />)
    expect(toJson(tree)).toMatchSnapshot()
  })
  
})