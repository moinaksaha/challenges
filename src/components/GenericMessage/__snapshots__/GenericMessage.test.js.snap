// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <GenericMessage /> Snapshot: Enzyme 1`] = `
<ThemeProvider
  theme={
    Object {
      "main": "#429845",
    }
  }
>
  <styled.div
    theme={
      Object {
        "main": "blue",
      }
    }
  >
    Some Message
  </styled.div>
</ThemeProvider>
`;
